Chess/board.cpp

```diff
@@ -90,3 +99,27 @@ void Board::initByColor(Color color)
         pieces[color].push_back(pawn);
     }
 }
+
+void Board::deletePiece(int row, int column, Color color)
+{
+    std::vector<Piece*> _pieces = getPieces(color);
+
+    int cnt = 0;
+    bool ok = 0;
+
+    for(auto pieceIndex : _pieces)
+    {
+        if(pieceIndex->getRow() == row && pieceIndex->getColumn() == column)
+        {
+            ok = 1;
+            break;
+        }
+        cnt++;
+    }
```

Comment:
Tot for-ul asta ar putea fi scos intr-o metoda privata separata care iti intoarce a cata piesa este in map.